src/mon/MgrStatMonitor.cc

```diff
@@ -66,9 +66,19 @@ void MgrStatMonitor::create_initial()
   encode(service_map, pending_service_map_bl, CEPH_FEATURES_ALL);
 }
 
+void MgrStatMonitor::clear_pool_availability(int64_t poolid)
+{
+  dout(20) << __func__ << dendl;
+  std::scoped_lock l(lock);
+  pending_pool_availability.erase(poolid);
+  pending_pool_availability.insert({poolid, PoolAvailability()});
+  dout(20) << __func__ << " cleared availability score for pool: " << poolid << dendl;
+}
```

Comment:
Used the logic suggested by Sam here: https://github.com/ceph/ceph/pull/63159#discussion_r2138356340. Seems to be working great, have tested it too!